mlflow/models/evaluation/default_evaluator.py

```diff
@@ -332,6 +349,37 @@ def __validate_artifacts(artifacts):
     )
 
 
+def _infer_artifact_type_and_ext(artifact_name, raw_artifact, exception_header):
+    """
+    This function performs type and file extension inference on the provided artifact
+    :param artifact_name: The name of the provided artifact
+    :param raw_artifact: The artifact object
+    :param exception_header: The header of any exceptions thrown in this function
+    :return: A tuple comprised of: a boolean representing whether or not raw_artifact is a path
+             to a file, an EvaluationArtifact (e.g. ImageEvaluationArtifact) class, a string
+             representing the appropriate file extension of the artifact.
+    """
+
+    # Given a path-like object: a string representation of path, or a pathlib.Path object
+    if isinstance(raw_artifact, str):
+        raw_artifact = pathlib.Path(raw_artifact)
+    if isinstance(raw_artifact, pathlib.Path):
+        if raw_artifact.is_file() and raw_artifact.suffix in EXT_TO_ARTIFACT_MAP.keys():
+            return True, EXT_TO_ARTIFACT_MAP[raw_artifact.suffix], raw_artifact.suffix
+        raise MlflowException(
+            f"{exception_header} produced an unsupported artifact '{artifact_name}' with path "
+            f"'{raw_artifact}'. The supported file extensions are: "
+            f"{', '.join(EXT_TO_ARTIFACT_MAP.keys())}"
+        )
+
+    # Given as other python object
+    return (
+        False,
+        TYPE_TO_ARTIFACT_MAP.get(type(raw_artifact), JsonEvaluationArtifact),
+        TYPE_TO_EXT_MAP.get(type(raw_artifact), ".json"),
+    )
```

Comment:
Can we create a `namedtuple` for this? `namedtuples` are more self-documenting and are easier to maintain.